src/adapters/listing-wizard/v1.ts

```diff
@@ -55,6 +59,51 @@ export class ListingWizardAdapterV1 extends Adapter {
     );
   }
 
+  /**
+   * Estimates the gas amount needed for creating new asset listing.
+   * @param universeId Universe ID.
+   * @param assetListingParams Listing params.
+   * @param listingTerms Listing terms.
+   */
+  async estimateCreateListingWithTerms(
+    universeId: BigNumberish,
+    assetListingParams: AssetListingParams,
+    listingTerms: IListingTermsRegistry.ListingTermsStruct,
+  ): Promise<BigNumber> {
+    const { encodedAssets, listingParams, maxLockPeriod, immediatePayout } = ListingHelper.prepareListingParams(
+      assetListingParams,
+      this.addressTranslator,
+    );
+
+    return this.contract.estimateGas.createListingWithTerms(
+      encodedAssets,
+      listingParams,
+      listingTerms,
+      maxLockPeriod,
+      immediatePayout,
+      universeId,
+    );
+  }
+
+  /**
+   * Creates new asset listings.
+   * @param listings List of listings.
+   */
+  async createListingsWithTerms(listings: CreateListingParams[]): Promise<ContractTransaction[]> {
+    const batches = await this.createBatches(listings);
+    const transactions: ContractTransaction[] = [];
+
+    for (const batch of batches) {
+      const tx = await this.contract.multicall(batch);
+      if (tx) {
+        await tx.wait();
```

Comment:
This waits for `tx` to mine. Is this really necessary?

Reply:
Maybe one solution would be to require user (backend) to generate an id for each listing and then when we batch them together, return `ContractTransaction` and list of associated ids. This way the backend could track transactions and map them to specific listings.